shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/impl/MySQLTCLStatementSQLVisitor.java

```diff
@@ -98,7 +108,32 @@ public ASTNode visitSavepoint(final SavepointContext ctx) {
     }
     
     @Override
-    public ASTNode visitXa(final XaContext ctx) {
-        return new MySQLXAStatement();
+    public ASTNode visitXaBegin(final XaBeginContext ctx) {
+        return new MySQLXABeginStatement();
```

Comment:
Is the `xid` been ignored?

Reply:
yes, xid is generated by TransactionManager, and will be managed in itself, i.e. per xid per connection

Reply:
Another way is pass xid to atomikos, but I couldn't find any possible interface to use. By the way JTA only provides begin, commit and rollback.

Reply:
yeah, because the JTA and XA are different protocol.

- JTA is the API used between the Application (AP) and Transaction Manager (TM) .
- XA is used between Transaction Manager (TM) and Resource Manager (RM) (such as database, message broker, ...) to coordinate during the 2PC commit.

So we can not use JTA here to handle the XA statments